scripts/build-layout-scales.js

```diff
@@ -0,0 +1,114 @@
+'use strict';
+
+const stripIndent = require('strip-indent');
+const mediaQueries = require('../src/media-queries');
+const allScales = require('../src/scales');
+
+function buildLayoutScales(config) {
+  const layoutScales = typeof config === 'object' ? config : Object.assign(allScales, config);
+  const variantGenerators = {};
+
+  variantGenerators.gutter = function (scales) {
+    let css = stripIndent(`
+      /**
+       * Apply column gutters to all columns in a grid by adding a <code>grid--gut{size}</code> modifier to the <code>grid</code> element. Class set includes <code>*-mm</code>, <code>*-ml</code>, and <code>*-mxl</code> variations to target screen sizes.
+       *
+       * @example
+       * <div class='grid grid--gut12'>
+       *   <div class='col col--2'>
+       *     <div class='border border--darken10'>col--2</div>
+       *   </div>
+       *   <div class='col col--4'>
+       *     <div class='border border--darken10'>col--4</div>
+       *   </div>
+       *   <div class='col col--6'>
+       *     <div class='border border--darken10'>col--6</div>
+       *   </div>
+       * </div>
+       * @memberof Grid
+       * @group
+       */`
+    );
+    css += scales.reduce((result, scale) => {
+      return result += stripIndent(`
+        .grid--gut${scale} {
+          margin-left: -${scale}px !important;
+        }
+      `);
+    }, '');
+    css += '/** @endgroup */\n';
+    css += scales.reduce((result, scale) => {
+      return result += stripIndent(`
+        .grid--gut${scale} > .col {
+          padding-left: ${scale}px !important;
+        }
+      `);
+    }, '');
+    css += '\n@media (' + mediaQueries['--m-screen'] + ') {\n';
+    css += scales.reduce((result, scale) => {
+      return result += stripIndent(`
+        .grid--gut${scale}-mm {
+          margin-left: -${scale}px !important;
+        }
+      `);
+    }, '');
+    css += scales.reduce((result, scale) => {
+      return result += stripIndent(`
+        .grid--gut${scale}-mm > .col {
+          padding-left: ${scale}px !important;
+        }
+      `);
+    }, '\n');
+    css += '\n}\n';
```

Comment:
This would probably be a little more clear & legible with an arrangement like:

```js
const medium = scales.reduce(..);
const mediumColumns = scales.reduce(..);
css += stripIndent(`
  @media (${mediaQueries['--m-screen']}) {
    ${medium}
    ${mediumColumns}
  }
`)
```

But maybe this is just quibbly.